src/components/TaxPayer/SpouseAndDependent.test.tsx

```diff
@@ -0,0 +1,250 @@
+import React from 'react'
+import { render, screen, fireEvent, waitFor } from '@testing-library/react'
+import { Provider } from 'react-redux'
+import userEvent from '@testing-library/user-event'
+
+import { SpouseInfo } from './SpouseAndDependent'
+import { store } from '../../redux/store'
+
+afterEach(async () => {
+  await waitFor(() => localStorage.clear())
+  jest.resetAllMocks()
```

Comment:
This is not working as expected. I need to properly mock `redux-persist`. Depending on the testing strategy moving forward, I can work on that next.